src/main/java/core/mate/academy/model/TruckProducer.java

```diff
@@ -0,0 +1,12 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<? extends Machine> get() {
+        Machine machine = new Truck();
```

Comment:
You are creating a `Machine` type object instead of a `Truck` type object. Since this is a `TruckProducer`, you should instantiate a `Truck` object directly.